text/0249-v2-experiments.md

```diff
@@ -408,3 +484,240 @@ Disadvantages:
   leads to duplicated classes between the 2 versions.
 
 **Verdict**: discarded as a worse version of option #5.
+
+# Appendix D - Alternatives for identifying stable/experimental builds
+
+We have the following options to pick from:
+
+- Different package names (`aws-cdk-lib` vs `aws-cdk-lib-experimental`).
+- Pre-release version tag
+- Different major versions
+
+We need to evaluate all of these, for each package manager, on the following criteria:
+
+- Can applications depend on an upwards unbounded range of *stable* builds? (important)
```

Comment:
"applications" - do you mean "libraries" here?

Reply:
I do not. We want application builders to migrate along with us to new CDK versions quickly. That is, I would want app builders to depend on ranges, not on point versions.

Reply:
Interesting. I feel like I have the *exact* opposite approach (you should only ever depend on pinned versions, and any upgrades should be done through a commit that you can revert if anything goes wrong),...

I feel like most tools agree it's better to pin versions, with how popular lock files have become. So maybe this is not an important point for apps? (I agree it's important for libs though)